core/src/main/java/jenkins/fingerprints/FileFingerprintStorage.java

```diff
@@ -322,10 +326,19 @@ private static String messageOfParseException(Throwable throwable) {
      * Deletes a directory if it's empty.
      */
     private void deleteIfEmpty(File dir) {
-        String[] r = dir.list();
-        if(r==null)     return; // can happen in a rare occasion
-        if(r.length==0)
-            dir.delete();
+        if (Files.isDirectory(dir.toPath())) {
+            try {
+                boolean isEmpty;
+                try (DirectoryStream<Path> directory = Files.newDirectoryStream(dir.toPath())) {
+                    isEmpty = !directory.iterator().hasNext();
+                }
+                if (isEmpty) {
+                    Files.delete(dir.toPath());
+                }
+            } catch (IOException e) {
+                throw new UncheckedIOException(e);
```

Comment:
Not sure about this. Depends on the caller.

Reply:
I made this log instead.